Libraries/LibWeb/Painting/SVGMaskable.cpp

```diff
@@ -87,7 +87,11 @@ RefPtr<Gfx::ImmutableBitmap> SVGMaskable::calculate_mask_of_svg(PaintContext& co
         DisplayListRecorder display_list_recorder(*display_list);
         display_list_recorder.translate(-mask_rect.location().to_type<int>());
         auto paint_context = context.clone(display_list_recorder);
-        paint_context.set_svg_transform(graphics_element.get_transform());
+        auto const& mask_element = as<SVG::SVGGraphicsElement const>(*paintable.dom_node());
+        paint_context.set_svg_transform(
+            graphics_element.get_transform()
+                .multiply(mask_element.get_transform().inverse().value())
+                .multiply(mask_element.element_transform()));
```

Comment:
could you add a comment with explanation why this transformation has to be applied?